src/components/ha-tooltip.ts

```diff
@@ -0,0 +1,25 @@
+import SlTooltip from "@shoelace-style/shoelace/dist/components/tooltip/tooltip.component";
+import styles from "@shoelace-style/shoelace/dist/components/tooltip/tooltip.styles";
+import { css } from "lit";
+import { customElement } from "lit/decorators";
+
+@customElement("ha-tooltip")
+export class HaTooltip extends SlTooltip {
+  static override styles = [
+    styles,
+    css`
+      :host {
+        --sl-tooltip-background-color: var(--secondary-background-color);
```

Comment:
Lets explicitly set the font color and font size too, so it will not use the color of the parent.